authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHadoopSQLPlugin.java

```diff
@@ -76,6 +85,297 @@ public Map<Privilege.Name, Set<AuthorizationPrivilege>> privilegesMappingRule()
+    }
+  }
+
+  /**
+   * Remove the SCHEMA, Need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` permissions.
+   */
+  private void doRemoveSchemaMetadataObject(AuthorizationMetadataObject authMetadataObject) {
+    Preconditions.checkArgument(
+        authMetadataObject.type() == SCHEMA, "The metadata object type must be SCHEMA");
+    Preconditions.checkArgument(
+        authMetadataObject.names().size() == 1, "The metadata object names must be 1");
+    if (RangerHelper.RESOURCE_ALL.equals(authMetadataObject.name())) {
+      // Delete metalake or catalog policies in this Ranger service
+      try {
+        List<RangerPolicy> policies = rangerClient.getPoliciesInService(rangerServiceName);
+        policies.stream()
+            .filter(RangerHelper::hasGravitinoManagedPolicyItem)
+            .forEach(rangerHelper::removeAllGravitinoManagedPolicyItem);
+      } catch (RangerServiceException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      List<List<String>> loop =
+          ImmutableList.of(
+              ImmutableList.of(authMetadataObject.name())
+              /** SCHEMA permission */
+              ,
+              ImmutableList.of(authMetadataObject.name(), RangerHelper.RESOURCE_ALL)
+              /** TABLE permission */
+              ,
+              ImmutableList.of(
+                  authMetadataObject.name(), RangerHelper.RESOURCE_ALL, RangerHelper.RESOURCE_ALL)
+              /** COLUMN permission */
+              );
+
+      for (int index = 0; index < loop.size(); index++) {
+        AuthorizationMetadataObject.Type type =
+            (index == 0
+                ? RangerHadoopSQLMetadataObject.Type.SCHEMA
+                : (index == 1
+                    ? RangerHadoopSQLMetadataObject.Type.TABLE
+                    : RangerHadoopSQLMetadataObject.Type.COLUMN));
+        AuthorizationMetadataObject authzMetadataObject1 =
+            new RangerHadoopSQLMetadataObject(
+                AuthorizationMetadataObject.getParentFullName(loop.get(index)),
+                AuthorizationMetadataObject.getLastName(loop.get(index)),
+                type);
+        removePolicyByMetadataObject(authzMetadataObject1);
+      }
+    }
+  }
+
+  /**
+   * Remove the TABLE, Need to remove these the relevant policies, `*.{table}`, `*.{table}.{column}`
+   * permissions.
+   */
+  private void doRemoveTableMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    List<List<String>> loop =
+        ImmutableList.of(
+            authzMetadataObject.names()
+            /** TABLE permission */
+            ,
+            Stream.concat(
+                    authzMetadataObject.names().stream(), Stream.of(RangerHelper.RESOURCE_ALL))
+                .collect(Collectors.toList())
+            /** COLUMN permission */
+            );
+
+    for (int index = 0; index < loop.size(); index++) {
+      AuthorizationMetadataObject.Type type =
```

Comment:
List<AuthorizationMetadataObject.Type> types = Lists.newArrayList(RangerHadoopSQLMetadataObject.Type.SCHEMA, RangerHadoopSQLMetadataObject.Type.TABLE, RangerHadoopSQLMetadataObject.Type.COLUMN);
types.get(index)

Reply:
OK, I refactor this code.